java-checks/src/test/files/checks/SynchronizationOnGetClassCheck.java

```diff
@@ -0,0 +1,32 @@
+public class MyClass {
+  public void doSomethingSynchronized() {
+    synchronized (getClass()) { // Noncompliant [[sc=19;ec=29]] {{Synchronize on the static class name instead.}}
+      // ...
+    }
+  }
+
+  public void doSomethingSynchronized2() {
+    synchronized (this.getClass()) { // Noncompliant [[sc=19;ec=34]] {{Synchronize on the static class name instead.}}
+      // ...
+    }
+  }
+}
+
+
+public final class FinalClassIsCompliant {
+  public void doSomethingSynchronized() {
+    synchronized (this.getClass()) { // Compliant
+      // ...
+    }
+  }
+}
+
+public class MemberSelect {
```

Comment:
No need for a separate class, named the method appropriately and put it as a member of the first class ?